src/main/java/edu/harvard/iq/dataverse/dataaccess/SwiftAccessIO.java

```diff
@@ -48,43 +48,48 @@
 
     private static final Logger logger = Logger.getLogger("edu.harvard.iq.dataverse.dataaccess.SwiftAccessIO");
 
-    public SwiftAccessIO() {
-        this((T)null);
-    }
-
-    public SwiftAccessIO(T dvObject) {
-        this(dvObject, null);
-    }
-
-    public SwiftAccessIO(T dvObject, DataAccessRequest req) {
-        super(dvObject, req);
-
+	public SwiftAccessIO() {
+		//Partially functional StorageIO object - constructor only for testing
+		super();
+	}
+	
+    public SwiftAccessIO(T dvObject, DataAccessRequest req, String driverId) {
+        super(dvObject, req, driverId);
+        readSettings();
         this.setIsLocalFile(false);
     }
     
-    public SwiftAccessIO(String swiftLocation) {
-        this((T)null);
+    public SwiftAccessIO(String swiftLocation, String driverId) {
+    	super(swiftLocation, driverId);
+    	readSettings();
         this.swiftLocation = swiftLocation;
+        this.setIsLocalFile(false);
     }
 
-    private Account account = null;
+    private void readSettings() {
+    	isPublicContainer = Boolean.parseBoolean(System.getProperty("dataverse.files." + this.driverId + ".isPublicContainer", "true"));
+        swiftFolderPathSeparator = System.getProperty("dataverse.files." + this.driverId + ".folderPathSeparator", "_");
+        swiftDefaultEndpoint = System.getProperty("dataverse.files." + this.driverId + ".defaultEndpoint");
+        tempUrlExpires = Integer.parseInt(System.getProperty("dataverse.files." + this.driverId + ".temporaryUrlExpiryTime", "60"));
+		
+	}
+
+	private Account account = null;
     private StoredObject swiftFileObject = null;
     private Container swiftContainer = null;
-    private boolean isPublicContainer = Boolean.parseBoolean(System.getProperty("dataverse.files.swift.isPublicContainer", "true"));
-    private String swiftFolderPathSeparator = System.getProperty("dataverse.files.swift.folderPathSeparator", "_");
-    private String swiftDefaultEndpoint = System.getProperty("dataverse.files.swift.defaultEndpoint");
+    private boolean isPublicContainer = true;
+    private String swiftFolderPathSeparator = "_";
+    private String swiftDefaultEndpoint = null;
 
     //for hash
     private static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
     
     //TODO: should this be dynamically generated based on size of file?
     //Also, this is in seconds
-    private static int TEMP_URL_EXPIRES = Integer.parseInt(System.getProperty("dataverse.files.swift.temporaryUrlExpiryTime", "60"));
+    private int tempUrlExpires = 60;
```

Comment:
Why has this been hardcoded? Is it not compatible w/ the new driver ID configuration format?

Reply:
The static value has been replaced by one specific to the swift store. It is set to a default of 60 and then in line 73 above is reset based on the property. (Nominally it's redundant to set it here and then set a default on line 73 as well...)